plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -7274,9 +7277,7 @@ public int compare(final UnmanagedInstanceTO.Disk disk1, final UnmanagedInstance
                 }
 
                 int extractInt(UnmanagedInstanceTO.Disk disk) {
-                    String num = disk.getLabel().replaceAll("\\D", "");
-                    // return 0 if no digits found
-                    return num.isEmpty() ? 0 : Integer.parseInt(num);
+                    return disk.getPosition() + (disk.getControllerUnit() * 100);
```

Comment:
@DK101010 I'm not sure if this formula is giving correct position/order for the disk.
I tried listUnamanagedInstance API for a VM with 2 data disks and results were incorrect with this change
![Screenshot from 2021-01-14 14-08-56](https://user-images.githubusercontent.com/153340/104565707-ac2fb400-5672-11eb-8d20-302442cd6565.png)

**With changes** - Root disk is showing up at second position in order
```
> list unmanagedinstances clusterid=2b35fefc-892a-4393-9ccf-30229c1a999b name=importtest
{
  "count": 1,
  "unmanagedinstance": [
    {
      "clusterid": "2b35fefc-892a-4393-9ccf-30229c1a999b",
      "cpucorepersocket": 1,
      "cpunumber": 1,
      "cpuspeed": 0,
      "disk": [
        {
          "capacity": 21474836480,
          "controller": "lsilogic",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-2000",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_61.vmdk",
          "label": "Hard disk 2",
          "position": 0
        },
        {
          "capacity": 2147483648,
          "controller": "ide",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-3001",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_62.vmdk",
          "label": "Hard disk 1",
          "position": 1
        },
        {
          "capacity": 5368709120,
          "controller": "lsilogic",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-2001",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_63.vmdk",
          "label": "Hard disk 3",
          "position": 1
        }
      ],
      "hostid": "fd9a2641-ce6e-456e-be78-4b1b63c2b2ac",
      "memory": 512,
      "name": "importtest",
      "nic": [
        {
          "adaptertype": "E1000",
          "id": "Network adapter 1",
          "macaddress": "02:00:44:4f:00:12",
          "networkname": "cloud.guest.2183.200.1-vSwitch1",
          "vlanid": 2183
        }
      ],
      "osdisplayname": "CentOS 4/5 or later (64-bit)",
      "osid": "centos64Guest",
      "powerstate": "PowerOn"
    }
  ]
}
```

**Without change** - Root disk at first place
```
> list unmanagedinstances clusterid=2b35fefc-892a-4393-9ccf-30229c1a999b name=importtest
{
  "count": 1,
  "unmanagedinstance": [
    {
      "clusterid": "2b35fefc-892a-4393-9ccf-30229c1a999b",
      "cpucorepersocket": 1,
      "cpunumber": 1,
      "cpuspeed": 0,
      "disk": [
        {
          "capacity": 2147483648,
          "controller": "ide",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-3001",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_62.vmdk",
          "label": "Hard disk 1",
          "position": 1
        },
        {
          "capacity": 21474836480,
          "controller": "lsilogic",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-2000",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_61.vmdk",
          "label": "Hard disk 2",
          "position": 0
        },
        {
          "capacity": 5368709120,
          "controller": "lsilogic",
          "controllerunit": 0,
          "datastorehost": "10.10.0.16",
          "datastorename": "4ea48dd0701d3385a5cbb8921ee1d1d4",
          "datastorepath": "/acs/primary/pr4385-t3314-vmware-65u2/pr4385-t3314-vmware-65u2-esxi-pri2",
          "datastoretype": "NFS",
          "id": "35-2001",
          "imagepath": "[4ea48dd0701d3385a5cbb8921ee1d1d4] importtest/importtest_63.vmdk",
          "label": "Hard disk 3",
          "position": 1
        }
      ],
      "hostid": "fd9a2641-ce6e-456e-be78-4b1b63c2b2ac",
      "memory": 512,
      "name": "importtest",
      "nic": [
        {
          "adaptertype": "E1000",
          "id": "Network adapter 1",
          "macaddress": "02:00:44:4f:00:12",
          "networkname": "cloud.guest.2183.200.1-vSwitch1",
          "vlanid": 2183
        }
      ],
      "osdisplayname": "CentOS 4/5 or later (64-bit)",
      "osid": "centos64Guest",
      "powerstate": "PowerOn"
    }
  ]
}

```

Reply:
Hi @shwstppr, the list is correct sorted by controller id and position. Works like I want. Your root disk is on controller Id 0 and position 1 and you have two disk on the same position. I'm a little bit confused. In my mind vmware puts the root disk in every case on controller id 0 and position 0.

Reply:
Hi @DK101010, I'm not sure why two disks have same position. To get this unmanaged VM. I just created a VM in cloudstack, added two data disks to it and then cloned the vm in vcenter.

Reply:
@shwstppr ahh ok you have two controller(ide, lsilogic), But in my opinion it makes no sense that you have one disk on one controller on position one. Also should increment the controller id, when you use more the one controller. And for this reason works this feature not correct. I think you found a other bug.

Reply:
Hi @shwstppr, I talked with my team about this behavior. First, your controller ids are correct. Increments are per controller type. For this reason is my implementation not better as the old one. Perhaps we could check in the sort algorithm if device type == scsi controller, but then it isn't no longer possible to boot from other controller types. It exist no safety way to find out which disk is bootable. The only way to get sure that  CS use the right volume is that the user indicates during the ingest process via rest call which volume is the right. Therefore I think we should close this pull request. What do you think?

Reply:
@DK101010 then is it better to add a new param in importUnamangedInstance API, something like `rootdiskid`? cc @rhtyd @PaulAngus

Reply:
@shwstppr yes, for example, but in my mind it should be optional for the user and when not set then works like now via sorting of disk names.